docs/guides/extension_points.rst

```diff
@@ -0,0 +1,180 @@
+
+For a more detailed comparison of content integration options, see `Options for Extending the edX Platform`_ in the *Open edX Developer's Guide*.
+
+.. _XBlock tutorial: https://edx.readthedocs.io/projects/xblock-tutorial/en/latest/
+.. _as a consumer: https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/lti_component.html
+.. _as a provider: https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/lti/
+.. _Options for Extending the edX Platform: https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/extending_platform/extending.html
+.. _custom JavaScript application: https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/extending_platform/javascript.html
+.. _external grader documentation: https://edx.readthedocs.io/projects/open-edx-ca/en/latest/exercises_tools/external_graders.html
+
+
+
+
+Platform Features (Integrating Python Code)
+===========================================
+
+If you wish to customize aspects of the learner or educator experiences, you'll most likely need to create a Python plugin that can be installed into the platform and which uses Python APIs to extend the platform's functionality.
+
+Most python plugins are enabled using one of two methods:
+
+1. A Python Entry point: the core Open edX platform provides a standard plugin loading mechanism in |openedx.core.lib.plugins|_ which uses `stevedore`_ to find all installed python packages that declare a specific "entry point" in their setup.py file. See the ``entry_points`` defined in edx-platform's own ``setup.py`` for examples.
+2. A Django setting: Some plugins require modification of Django settings, which is typically done by editing ``/edx/etc/lms.yml`` (in Production) or ``edx-platform/lms/envs/private.py`` (on Devstack).
+
+.. |openedx.core.lib.plugins| replace:: ``openedx.core.lib.plugins``
+.. _openedx.core.lib.plugins: https://github.com/edx/edx-platform/blob/master/openedx/core/lib/plugins.py
+.. _stevedore: https://pypi.org/project/stevedore/
+
+Here are the different integration points that python plugins can use:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 15 10 75
+
+   * - Plugin Type
+       (and entry point or setting)
+     - Status
+     - Details
+   * - Django App Plugin (``lms.djangoapp`` and ``cms.djangoapp``)
+     - Adopt, Stable
+     - A "Django app plugin" is a self-contained Django `Application`_ that can define models (MySQL tables), new REST APIs, signal listeners, asynchronous tasks, and more. Even some parts of the core platform are implemented as Django app plugins, for better separation of concerns (``announcements``, ``credentials``, ``grades``, etc.) Read the `Django app plugin documentation`_ to learn more.
+
+       Plugins can also inject custom data into django template contexts, to affect standard pages delivered by the core platform. See `Plugin Contexts`_ to learn more.
```

Comment:
FYI: We are in process of moving this to edx-django-utils.  See https://github.com/edx/edx-django-utils/pull/48

Reply:
> FYI: We are in process of moving this to edx-django-utils. See edx/edx-django-utils#48

OK, cool. Could you please update this doc at the same time, to point to that new home for the functionality?

Reply:
Oops.  I thought this hadn't landed yet because I got a 404 on: https://github.com/open-craft/edx-platform/blob/master/docs/guides/extension_points.rst
I didn't realize I was searching in the wrong org.  We can update the following accordingly:
https://github.com/edx/edx-platform/blob/master/docs/guides/extension_points.rst
Thanks.